plugins/out_stackdriver/stackdriver_conf.c

```diff
@@ -414,68 +455,61 @@ struct flb_stackdriver *flb_stackdriver_conf_create(struct flb_output_instance *
         }
     }
 
-    if (flb_sds_cmp(ctx->resource, "k8s_container",
-                    flb_sds_len(ctx->resource)) == 0 ||
-        flb_sds_cmp(ctx->resource, "k8s_node",
-                    flb_sds_len(ctx->resource)) == 0 ||
-        flb_sds_cmp(ctx->resource, "k8s_pod",
-                    flb_sds_len(ctx->resource)) == 0) {
+    set_resource_type(ctx);
```

Comment:
For readability, I think we should catch the error of failing to assign the resource type here instead of using a void return. We're now delegating the error check of resource type not supported in the function `resource_api_has_required_labels` but I think it's better to catch up the unsupported resource type in the function `set_resource_type`.

Reply:
Would we want to move the warning to `set_resource_type` as well? The way it currently is, "no validation applied to resource_labels for set resource type" will only appear if the user is using the new API, since we check for the list size first in `resource_api_has_required_labels`.  If we change the logic, that path will never be taken, since we will check the response from `set_resource_type` first. We could make the warning message more general (i.e. "no resource label validation is supported for set resource type"), but I am wondering if it would be bothersome for existing users or for resource types that might not really need validation (e.g. global?). WDYT?

Reply:
SG. I added a comment there.